bigquery/google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Cursor for the Google BigQuery DB-API."""
+
+import collections
+import uuid
+
+import six
+
+from google.cloud.bigquery.dbapi import _helpers
+from google.cloud.bigquery.dbapi import exceptions
+
+
+_ARRAYSIZE_DEFAULT = 20
+
+
+class Cursor(object):
+    """DB-API Cursor to Google BigQuery.
+
+    :type: :class:`~google.cloud.bigquery.dbapi.Connection`
+    :param connection: A DB-API connection to Google BigQuery.
+    """
+    def __init__(self, connection):
+        self.connection = connection
+        self.description = None
+        self.rowcount = -1
+        self.arraysize = None
+        self._query_data = None
+        self._page_token = None
+        self._has_fetched_all_rows = True
+
+    def close(self):
+        """No-op."""
+
+    def _set_description(self, schema):
+        """Set description from schema.
+
+        :type: list of :class:`~google.cloud.bigquery.schema.SchemaField`
+        """
+        if schema is None:
+            self.description = None
+            return
+
+        desc = []
+        for field in schema:
+            desc.append(tuple([
```

Comment:
Can you use a NamedTuple that matches the definition in PEP 249? https://www.python.org/dev/peps/pep-0249/#description